src/playback/internal/playbackcontroller.cpp

```diff
@@ -600,6 +600,11 @@ void PlaybackController::setupSequenceTracks()
     partList.onItemRemoved(this, [this](const Part* part) {
         removeTrack(part->id());
     });
+
+    partList.onItemChanged(this, [this](const Part* part) {
+        addTrack(part->id(), part->partName().toStdString());
+        removeTrack(part->id());
```

Comment:
Isn't that the wrong way round and should remove first and add last?

Reply:
Yes, that's what doesn't make sense. The other way doesn't work. When you do it the other way, the instrument gets removed, but the last instrument in the mixer gets duplicated